Framework/Indexing/src/SpectrumNumberTranslator.cpp

```diff
@@ -57,12 +91,12 @@ SpectrumNumberTranslator::makeIndexSet(SpectrumNumber min,
   static_cast<void>(m_spectrumNumberToPartition.at(min));
   static_cast<void>(m_spectrumNumberToPartition.at(max));
 
-  // The order of spectrum numbers can be arbitrary so we need to iterate.
+  std::call_once(m_mapSetup, &SpectrumNumberTranslator::setupMaps, this);
   std::vector<size_t> indices;
```

Comment:
Maybe, but I would usually wait for profiling results. Since this method is not yet in widespread use, I prefer keeping things simple.